master/buildbot/statistics/capture.py

```diff
@@ -38,6 +42,7 @@ def _defaultContext(self, msg):
             "build_number": str(msg['number'])
         }
 
+    @abc.abstractmethod
     def consume(self, routingKey, msg):
```

Comment:
When using **abc** module, you don't need to use `NotImplementedError` exception as it is the purpose of that module :)
Just add a docstring or a `pass` statement as you did below (_retValParams)